src/core/shards/casting.cpp

```diff
@@ -560,6 +560,25 @@ struct StringToBytes {
   }
 };
 
+struct ImageToBytes {
+  static SHTypesInfo inputTypes() { return CoreInfo::ImageType; }
+  static SHTypesInfo outputTypes() { return CoreInfo::BytesType; }
+
+  SHVar activate(SHContext *context, const SHVar &input) {
+    auto pixsize = 1;
+    if ((input.payload.imageValue.flags & SHIMAGE_FLAGS_16BITS_INT) == SHIMAGE_FLAGS_16BITS_INT)
```

Comment:
Can you move this to a helper function, the same code is also used below in Resize::activate

Reply:
sadly it's a bigger refactor as it's done in many places actually 😅

Reply:
done but will need to catch more locations